src/kernels/jupyter/session/jupyterSession.ts

```diff
@@ -234,7 +235,7 @@ export class JupyterSession extends BaseJupyterSession {
 
         // Create our session options using this temporary notebook and our connection info
         const sessionOptions: Session.ISessionOptions = {
-            path: backingFile?.filePath || `${uuid()}.ipynb`, // Name has to be unique
+            path: backingFile?.filePath || generateBackingIPyNbFileName(this.resource), // Name has to be unique
```

Comment:
Now all paths generate have a predictable name & we can strip out the unnecessary guff and have a meaningful and prettier display name in the kernel picker, previously we'd just display a guid.